docs/install.rst

```diff
@@ -0,0 +1,210 @@
+
+    $ pip install horovod[tensorflow,keras,pytorch,mxnet,spark]
+
+This is useful when building Horovod as part of a larger collection of dependencies at once, relying on the pip
+compiler to determine the correct install order.
+
+TensorFlow
+~~~~~~~~~~
+
+To ensure that Horovod is built with TensorFlow support enabled:
+
+.. code-block:: bash
+
+    $ HOROVOD_WITH_TENSORFLOW=1 pip install horovod[tensorflow]
+
+To skip TensorFlow, set ``HOROVOOD_WITHOUT_TENSORFLOW=1`` in your environment.
+
+PyTorch
+~~~~~~~
+
+To ensure that Horovod is built with PyTorch support enabled:
+
+.. code-block:: bash
+
+    $ HOROVOD_WITH_PYTORCH=1 pip install horovod[pytorch]
+
+To skip PyTorch, set ``HOROVOOD_WITHOUT_PYTORCH=1`` in your environment.
+
+MXNet
+~~~~~
+
+To ensure that Horovod is built with MXNet support enabled:
+
+.. code-block:: bash
+
+    $ HOROVOD_WITH_MXNET=1 pip install horovod[mxnet]
+
+To skip MXNet, set ``HOROVOOD_WITHOUT_MXNET=1`` in your environment.
+
+Keras
+~~~~~
+
+Standalone Keras support is currently only available for the TensorFlow backend.
+
+To ensure that Horovod is built with Keras support available:
+
+.. code-block:: bash
+
+    $ HOROVOD_WITH_TENSORFLOW=1 pip install horovod[tensorflow,keras]
+
+There are no plugins built for Keras, but the TensorFlow plugin must be enabled in order to use Horovod with Keras.
+
+Spark
+~~~~~
+
+Horovod can be used with Spark in combination with any of the frameworks above.
+
+To ensure Horovod has all the necessary requirements in order to run on top of Spark:
+
+.. code-block:: bash
+
+    $ pip install horovod[spark]
+
+Controllers
+-----------
+
+The controller is used for coordinating work between Horovod processes (determining which tensors to process). We
+provide controller implementations for both MPI and Gloo. By default, Horovod will attempt to build support for both
+of them. At least one must be enabled for Horovod to install successfully.
+
+MPI
+~~~
+
+MPI is the original controller for Horovod.  It uses ``mpirun`` to launch worker processes (``horovodrun`` will use
+``mpirun`` under the hood when using MPI).
+
+* To force Horovod to install with MPI support, set ``HOROVOD_WITH_MPI=1`` in your environment.
+* To force Horovod to skip building MPI support, set ``HOROVOD_WITHOUT_MPI=1``.
+
+If both MPI and Gloo are enabled in your installation, then MPI will be the default controller.
+
+Gloo
+~~~~
+
+Gloo is a more recent controller for Horovod that does not require additional dependencies besides CMake to install.
+
+When used as a controller in combination with NCCL, Gloo performs almost identically to MPI on standard benchmarks.
+
+* To force Horovod to install with Gloo support, set ``HOROVOD_WITH_GLOO=1`` in your environment.
+* To force Horovod to skip building MPI support, set ``HOROVOD_WITHOUT_GLOO=1``.
+
+Gloo mode uses ``horovodrun`` to launch worker processes.
+
+Gloo will be required for the upcoming elastic / fault tolerant API for Horovod.
```

Comment:
I would drop upcoming here, otherwise we would have to come back here and remove it when elastic is there.

Reply:
Good point.